### PR TITLE
feat: adding possibility to pass an object as parameter to customLevels

### PR DESCRIPTION
**Pull request by ademarsj**
Hello, i made the changes to make possible pass in code as object or string the customLevels prop, it's still compatible with the CLI passing as string parameter.  I already wrote the tests too and keeps on the 100% coverage and asserted.

Is relative to #399 .

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -669,3 +671,64 @@ function autoEnd (stream, eventName) {
...
+  if (typeof cLevels === 'string') {
+    return cLevels
+      .split(',')
+      .reduce((agg, value, idx) => {
+        const [levelName, levelIdx = idx] = value.split(':')
+        agg[levelIdx] = levelName.toUpperCase()
+        return agg
+      },
+      { default: 'USERLVL' })
+  } else if (typeof cLevels === 'object') {
```
Comment (jsumners) [inline lib/utils.js:694]: An instance of `[]` would pass this check. The correct check is:
```suggestion
  } else if (Object.prototype.toString.call(cLevels) === '[object Object]') {
```

File: lib/utils.js [hunk 2]
```diff
@@ -669,3 +671,64 @@ function autoEnd (stream, eventName) {
...
 }
+
+/**
+ * Returns a object which will be used to handle custom levels
+ *
+ * @param {string|object} cLevels Custom Levels can be object or string
+ *
+ * @returns {object} A object like level: level_name
+ */
+function handleCustomlevelsOpts (cLevels) {
```
Comment (jsumners) [inline lib/utils.js:682]: Thank you for providing jsdoc for the new function. However, the documentation should provide some real value over what the function signature already provides:

```suggestion
/**
 * Parse a CSV string or options object that specifies
 * configuration for custom levels.
 *
 * @param {string|object} cLevels An object mapping standard level
 * names to new level names, e.g. `{ info: 'information') }`, or a
 * CSV string in the format `level:new_name`, e.g.
 * `info:information,debug:devel`.
 *
 * @returns {object} A object mapping known level names to customized
 * level names.
 */
function handleCustomlevelsOpts (cLevels) {
```

File: lib/utils.js [hunk 3]
```diff
@@ -669,3 +671,64 @@ function autoEnd (stream, eventName) {
...
+  }
+}
+
+/**
+ * Returns a object which will be used to handle custom levels
+ *
+ * @param {string|object} cLevels Custom Levels can be object or string
+ *
+ * @returns {object} A object like level_name: level
+ */
```
Comment (jsumners) [inline lib/utils.js:712]: Please adjust this docblock in the same manner as my suggestion on the other one.

**Review by jsumners — CHANGES_REQUESTED** [review]
Looking good. I have some requested changes, and the documentation around the feature should also be updated.

File: lib/utils.js (hunk 1 above)
Comment (ademarsj) [reply lib/utils.js:694]: True, i never did like that, i will change both checks, thank you.

File: lib/utils.js (hunk 2 above)
Comment (ademarsj) [reply lib/utils.js:682]: Will be done.

File: lib/utils.js (hunk 3 above)
Comment (ademarsj) [reply lib/utils.js:712]: Ok

**Comment by ademarsj** [comment]
I appreciate the request and comments, i fixed them, but i got some doubts/points:
- I believe that you just give me an example of the method docs, so i changed a little to something more strict to the function returns
- I noticed that in `handleCustomlevelNamesOpts` method when passing an object as parameter the return is practically the same object that is passed, but i didn't remove the logic because of the lower case transformation, do you think it's appropriate to leave it that way ? Or just return the object as it came (expecting that the user will always pass an object with properties names already in lower case), and pass to lower case only in string mode ?
- I added the docs to the PrettyOptions with examples, I hope I made right

**Review by jsumners — CHANGES_REQUESTED** [review]

**Comment by jsumners** [comment]
> * I noticed that in `handleCustomlevelNamesOpts` method when passing an object as parameter the return is practically the same object that is passed, but i didn't remove the logic because of the lower case transformation, do you think it's appropriate to leave it that way ? Or just return the object as it came (expecting that the user will always pass an object with properties names already in lower case), and pass to lower case only in string mode ?

I would review the `git blame` for the implementation and the PRs that introduced it. The discussion therein should answer these questions.

> * I added the docs to the PrettyOptions with examples, I hope I made right

I don't understand. What is `PrettyOptions`?

**Comment by ademarsj** [comment]
> I would review the `git blame` for the implementation and the PRs that introduced it. The discussion therein should answer these questions.

I did it, in this PR,  as it was just passed as a string it always will transform to lower case to make the attributes names to the object returned, now it will receive an object, if the object received it's already with lower case in his attributes names the for loop is useless, once it will return an object same as received.

> I don't understand. What is `PrettyOptions`?
The name of the Interface in index.d.ts which describes the pino-pretty options.

I will fix the English errors that you appointed soon.

**Comment by ademarsj** [comment]
Done, sorry for the bad english.

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Comment by jsumners** [comment]
> Done, sorry for the bad english.

No worries. That's what reviews are for.

**Pull request merged**